common/proof.go

```diff
@@ -0,0 +1,61 @@
+package common
+
+import (
+	"errors"
+
+	ethtypes "github.com/ethereum/go-ethereum/core/types"
+	"github.com/ethereum/go-ethereum/rlp"
+	"github.com/zeta-chain/zetacore/common/ethereum"
+)
+
+// ErrInvalidProof is a error type for invalid proofs embedding the underlying error
+type ErrInvalidProof struct {
+	Err error
+}
+
+func NewErrInvalidProof(err error) ErrInvalidProof {
+	return ErrInvalidProof{
+		Err: err,
+	}
+}
+
+func (e ErrInvalidProof) Error() string {
+	return e.Err.Error()
+}
+
+// IsErrorInvalidProof returns true if the error is an ErrInvalidProof
+func IsErrorInvalidProof(err error) bool {
+	return errors.As(err, &ErrInvalidProof{})
+}
+
+// NewEthereumProof returns a new Proof containing an Ethereum proof
+func NewEthereumProof(proof ethereum.Proof) *Proof {
```

Comment:
Do you want to change the parameter type to `proof *ethereum.Proof` as I saw you had to deference the proof in your test `common.NewEthereumProof(*txProof)`.

Reply:
Changed `NewEthereumProof` to receive a pointer, makes more sense